action/protocol/staking/handlers.go

```diff
@@ -95,38 +91,11 @@ func (p *Protocol) handleCreateStake(ctx context.Context, act *action.CreateStak
 		return log, nil, csmErrorToHandleError(candidate.Owner.String(), err)
 	}
 
-	// update bucket pool
-	if err := csm.DebitBucketPool(act.Amount(), true); err != nil {
-		return log, nil, &handleError{
-			err:           errors.Wrapf(err, "failed to update staking bucket pool %s", err.Error()),
-			failureStatus: iotextypes.ReceiptStatus_ErrWriteAccount,
-		}
-	}
-
-	// update staker balance
-	if err := staker.SubBalance(act.Amount()); err != nil {
-		return log, nil, &handleError{
-			err:           errors.Wrapf(err, "failed to update the balance of staker %s", actionCtx.Caller.String()),
-			failureStatus: iotextypes.ReceiptStatus_ErrNotEnoughBalance,
-		}
-	}
-	// put updated staker's account state to trie
-	if err := accountutil.StoreAccount(csm.SM(), actionCtx.Caller, staker); err != nil {
-		return log, nil, errors.Wrapf(err, "failed to store account %s", actionCtx.Caller.String())
-	}
-
```

Comment:
i think make these 3 funcs calls into your `createBucket` new funcs, but should not include `fetchCaller(ctx, csm, act.Amount())` above
b/c if you include, it might end up with different error return. For example, now it errs at L102 `ReceiptStatus_ErrWriteAccount`, but original code could err out at L91 `ReceiptStatus_ErrInvalidBucketAmount`

Reply:
okay, it's possible to cause failure of fullsync on paper